### ISSUE TITLE
The specified module could not be found - oracledb.node

### ISSUE THREAD

**Issue by ZachMoreno** [oracle/node-oracledb#651]
Thanks in advance for any & all help. I'm happy to provide any other information that will help.

1. What error(s) you are seeing?
[Output gist](https://gist.github.com/ZachMoreno/b97d9178745615edca07b9ee57d7a06a)

2. What *exact* command caused the problem (e.g. what command did you try to install with)?  Who were you logged in as?
`node api.js`
Administrator

3. What environment variables did you set?  How *exactly* did you set them?
Computer > Properties > Advanced system settings > Environment Variables >System variables
`PATH`, `OCI_INC_DIR` & `OCI_LIB_DIR`

4. What is your version of Node.js?
v6.10.0

5. What is your version of the Oracle client (e.g. Instant Client)?  How was it installed?  Where it is installed?
12.1.0.2.0
installed from [Oracle downloads page](http://www.oracle.com/technetwork/database/features/instant-client/index-097480.html)
installed @ `C:\Oracle\instantclient`

6. What is your OS and version?
Windows Server 2012 R2

7. What compiler version did you use?
Visual Studio 2017 Community with Visual C++ Build Tools

![Node-Oracledb-Issue](https://www.coastal.ca.gov/images/node-oracledb-issue.png)

**Comment by ZachMoreno** [comment]
Followed all solutions mentioned in #69 & #382 & still not working.

**Comment by cjbj** [comment]
You have the correct MS RTL?

**Comment by ZachMoreno** [comment]
Please forgive my ignorance, I'm not 100% sure I know what you're referring to. [Microsoft Run-Time Library](https://msdn.microsoft.com/en-us/library/windows/hardware/ff563638(v=vs.85).aspx)?

**Comment by ZachMoreno** [comment]
I can't find any mention in the [README](https://github.com/oracle/node-oracledb/blob/master/README.md), [Install guide](https://github.com/oracle/node-oracledb/blob/master/INSTALL.md), or [Docs](https://github.com/oracle/node-oracledb/blob/master/doc/api.md) of MS RTL.

**Comment by cjbj** [comment]
Yes.

**Comment by ZachMoreno** [comment]
Yes.
![2017 Redistributable](https://coastal.ca.gov/images/2017-redistributable.PNG)

**Comment by kubo** [comment]
In [Node-oracledb Installation on Windows](https://github.com/oracle/node-oracledb/blob/master/INSTALL.md#instwin)
> The Oracle client 12.1 requires the [Visual Studio 2010 redistributable](https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads#bookmark-vs2010).

**Comment by ZachMoreno** [comment]
Thank you both so much!!
After getting [Microsoft Visual C++ 2010 Express x64 Redistributable - 10.0.40219](https://www.microsoft.com/en-us/download/details.aspx?id=26999) installed, node 🔥ed  right up. Are there plans to support a C++ runtime for Windows that's not past its [Mainstream Support End Date](https://support.microsoft.com/en-us/lifecycle/search/14048)?

**Comment by cjbj** [comment]
Oracle 12.2 needs the 2013 redistributable, I understand.

**Comment by soorajssv** [comment]
I faced similar issue and though everything was in place, I was getting the the specified module could not be found error and was struggling for 3 weeks. Using dependency walker, I found that the OCI dll was being referred from the directory c:\windows\system32 on my windows 7 box. I am not sure why the oci.dll was present in system32 folder. I renamed the oci.dll uder system32 folder to _oci.dll and everything worked fine thereafter. Hope this helps
